src/windows.rs

```diff
@@ -82,53 +93,95 @@ fn wide<T: AsRef<OsStr>>(input: T) -> Vec<u16> {
 
 /// Performs an operation on a specified file.
 ///
-/// <https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutew>
+/// <https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecuteexw>
 #[allow(non_snake_case)]
 #[cfg(feature = "shellexecute-on-windows")]
-pub unsafe fn ShellExecuteW(
-    hwnd: isize,
-    lpoperation: *const u16,
-    lpfile: *const u16,
-    lpparameters: *const u16,
-    lpdirectory: *const u16,
-    nshowcmd: i32,
-) -> std::io::Result<()> {
-    let hr = ffi::ShellExecuteW(
-        hwnd,
-        lpoperation,
-        lpfile,
-        lpparameters,
-        lpdirectory,
-        nshowcmd,
-    );
-
-    // ShellExecuteW returns > 32 on success
-    // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutew#return-value
-    if hr > 32 {
+pub unsafe fn ShellExecuteExW(info: *mut ffi::SHELLEXECUTEINFOW) -> std::io::Result<()> {
+    // ShellExecuteExW returns TRUE (i.e 1) on success
+    // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecuteexw#remarks
+    if ffi::ShellExecuteExW(info) == 1 {
         Ok(())
     } else {
         Err(std::io::Error::last_os_error())
     }
 }
 
 #[cfg(feature = "shellexecute-on-windows")]
+#[allow(non_snake_case)]
 mod ffi {
-    /// Activates the window and displays it in its current size and position.
+    /// Activates and displays a window.
+    /// If the window is minimized, maximized, or arranged, the system restores it to its original size and position.
+    /// An application should specify this flag when displaying the window for the first time.
     ///
     /// <https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showwindow>
-    pub const SW_SHOW: i32 = 5;
+    pub const SW_SHOWNORMAL: i32 = 1;
+
+    /// Null-terminated UTF-16 encoding of `explore`.  
+    pub const EXPLORE: *const u16 = [101, 120, 112, 108, 111, 114, 101, 0].as_ptr();
+
+    /// Null-terminated UTF-16 encoding of `folder`.  
+    pub const FOLDER: *const u16 = [102, 111, 108, 100, 101, 114, 0].as_ptr();
+
+    // Taken from https://docs.rs/windows-sys/latest/windows_sys/
+    #[repr(C)]
+    #[cfg(not(target_arch = "x86"))]
+    pub struct SHELLEXECUTEINFOW {
```

Comment:
The only difference here seems to be the `packed(1)` attribute.

I think this can be written like so:

```
#[cfg_attr(target_arch = "x86", repr(packed(1))]
```

I am not sure if `repr` stacks so that `repr(C)` can be followed by `repr(packed(1)` to produce `repr(C, packed(1))`, and if not the case, it should be possible to go with:

```
#[cfg_attr(target_arch = "x86", repr(C, packed(1))]
#[cfg_attr(not(target_arch = "x86"), repr(C)]
```

Reply:
resolved